### DISCUSSION TITLE
How to get plugins changelog

### DISCUSSION THREAD

**Discussion by LokiNaBoki** [wbthomason/packer.nvim#653] [Q&A] [answered]
When you update your plugins there is a floating window with all commits since last update for every plugin. When you run `PackerSync` with headless nvim there isn't any window. My question is can I somehow get this "changelog" either by calling some Packer function or somehow redirecting it to a file?

**Comment by wbthomason** [comment, marked as answer]
This would be an interesting feature to add, but isn't something `packer` currently supports - the changelogs are ephemeral and only displayed when you're running non-headless `packer`. You can, of course, view the `git log` of the relevant plugins manually yourself, but this is not built into `packer` currently. If you'd like to add a feature like this, I'd be happy to help guide a PR.

**Reply by gegoune** [reply]
I remember vaguely mentioning it before, vim-plug has PlugDiff command which shows diff for last update. Would be super nice to have something like that in Packer as well.

**Reply by wbthomason** [reply]
@gegoune there is a way to view the diff if you're not in headless mode; I believe @LokiNaBoki's question is only pertaining to headless operation?

**Reply by gegoune** [reply]
Oh, is there? How exactly? Do you mean pressing `d` with cursor on commit in Packer's window? I think @LokiNaBoki's question and my suggestion/FR are covering different scenario:

Do one of the following:
  - headlessly run `PackerSync`, close nvim
  - run `PackerSync` or `PackerUpdate`, close Packer's window
 
Once window/nvim is closed there is no way to see what its content was.

Now, imagine there is `PackerDiff`, or `PackerShowLastUpdateStatus` (ugly, only to demonstrate its exact meaning), which user can run in any of above situations and recall last Packer's window.

I needed it number of times when few plugins got updated and I was changing my configuration by reading change log, I have reconfigured one plugin and cycled my nvim to apply that configuration only to realise there were more plugins updated but I had no easy way of figuring out what they were (and what changes were) (except for checking `git log` for dozens of plugins).

**Reply by wbthomason** [reply]
Right - does `PlugDiff` work *after* you close Vim from an update? If so, maybe we can easily add such a feature.

**Reply by gegoune** [reply]
I really can't remember, I am sorry. It's been such a long time since I used it.

**Reply by wbthomason** [reply]
No worries! I don't have the bandwidth to prioritize this feature right now, but I'll try to look into what `plug` does at some future point.
